### PR TITLE
eclass/ros-catkin: first changes to install everything in /opt/ros

### PR DESCRIPTION
**Pull request by Alessandro-Barbieri**
Thanks to @allenh1

### CODE REVIEW

File: eclass/ros-catkin.eclass [hunk 1]
```diff
@@ -163,16 +163,19 @@ ros-catkin_src_prepare() {
 	append-cxxflags '-std=c++11'
 }
 
+ROSPREFIX="/opt/ros/${ROSDISTRO}"
```
Comment (aballier) [inline eclass/ros-catkin.eclass]: We can't move everything like that: imagine you rebuild one package that expects its dependencies to be in /opt/ros while those have been built before this eclass change; package will fail to find its dependencies

Also, ROSDISTRO will be undefined until all ebuilds have been updated, and a global scope ROSPREFIX would prevent from having the same package built for 2 or more distros

File: eclass/ros-catkin.eclass [hunk 2]
```diff
@@ -163,16 +163,19 @@ ros-catkin_src_prepare() {
...
 # Internal decoration of cmake-utils_src_configure to handle multiple python installs.
 ros-catkin_src_configure_internal() {
 	if [ -n "${CATKIN_DO_PYTHON_MULTIBUILD}" ] ; then
 		local sitedir="$(python_get_sitedir)"
+		local sitedir="${sitedir/\/usr\//}"
 		local mycmakeargs=(
 			"${mycmakeargs[@]}"
 			-DPYTHON_EXECUTABLE="${PYTHON}"
-			-DPYTHON_INSTALL_DIR="${sitedir#${EPREFIX}/usr/}"
+			-DPYTHON_INSTALL_DIR="${sitedir#${EPREFIX}}"
```
Comment (aballier) [inline eclass/ros-catkin.eclass]: Does this change make any actual difference ?

File: eclass/ros-catkin.eclass [hunk 3]
```diff
@@ -192,8 +195,15 @@ ros-catkin_src_configure_internal() {
 # @DESCRIPTION:
 # Configures a catkin-based package.
 ros-catkin_src_configure() {
-	export CATKIN_PREFIX_PATH="${EPREFIX}/usr"
-	export ROS_ROOT="${EPREFIX}/usr/share/ros"
+	if [ -f "${ROSPREFIX%/}/setup.bash" ]; then
```
Comment (aballier) [inline eclass/ros-catkin.eclass]: What package would install this file?

File: eclass/ros-catkin.eclass (hunk 1 above)
Comment (Alessandro-Barbieri) [inline eclass/ros-catkin.eclass]: What's the problem? You can't have a working ros now. (If yes tell me how)
I can remove the `ROSDISTRO` variable

File: eclass/ros-catkin.eclass (hunk 2 above)
Comment (Alessandro-Barbieri) [inline eclass/ros-catkin.eclass]: I don't think

**Comment by allenh1** [comment]
I'd recommend holding off on reviewing this for the time being, as I'm still iterating on the design.

File: eclass/ros-catkin.eclass (hunk 1 above)
Comment (aballier) [inline eclass/ros-catkin.eclass]: it works fine here

as for the problem with changing this like that, the one I mentioned is a quite major one

File: eclass/ros-catkin.eclass (hunk 2 above)
Comment (allenh1) [inline eclass/ros-catkin.eclass]: I does not in this case. There's a slight work around I'm still iterating on that deals with this variable.

File: eclass/ros-catkin.eclass (hunk 3 above)
Comment (allenh1) [inline eclass/ros-catkin.eclass]: This comes from catkin.

Comment (aballier) [inline eclass/ros-catkin.eclass]: @mgorny what do you think about sourcing this ? shall it use a subshell or something ?

**Comment by aballier** [comment]
Note: I'm in the process of removing multi-python support from ROS; it can't really work with the way rospack/roslib & co use python. This should simplify the eclass and ease defining a python version for each ros distro.

File: eclass/ros-catkin.eclass (hunk 3 above)
Comment (allenh1) [inline eclass/ros-catkin.eclass]: Please don't worry about this for the moment -- I'm still trying to figure things out and am trying to find a better way to do this.

**Pull request closed without merging**